File: src/scoreboard-client/lib/api/meta.ts
import { getJson } from "../http";
import type { MetaResponse } from "../dtos/api/meta";
import type { ScoreScope } from "../score_scope";

export function meta(scope: ScoreScope): Promise<MetaResponse> {
  return getJson<MetaResponse>(`/api/meta?scope=${scope}`);
}

File: src/scoreboard-server/scoreboard_server/services/api/meta.py
from __future__ import annotations

from scoreboard_server.cores.leaderboard import build_meta_payload
from scoreboard_server.db.repository import ScoreboardStore
from scoreboard_server.dtos.api.meta import MetaResponse


async def meta_response(
    store: ScoreboardStore, *, scope: str = "official"
) -> MetaResponse:
    entries = await store.list_latest_scores_for_space(
        is_tmp=scope == "non_official"
    )
    return build_meta_payload(entries, scope=scope)
